src/devices/Tcs3472x/Tcs3472x.cs

```diff
@@ -0,0 +1,215 @@
+                timeByte = Math.Clamp(timeByte, 0, 255);
+                _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ATIME), (byte)timeByte });
+                _integrationTimeByte = (byte)timeByte;
+            }
+            else
+            {
+                if (!_isLongTime)
+                {
+                    SetConfigLongTime(true);
+                }
+                _isLongTime = true;
+                var timeByte = (int)(0x100 - (timeSeconds / 0.029));
+                timeByte = Math.Clamp(timeByte, 0, 255);
+                _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.WTIME), (byte)timeByte });
+                _integrationTimeByte = (byte)timeByte;
+            }
+        }
+
+        private void SetConfigLongTime(bool setLong)
+        {
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.CONFIG), setLong ? (byte)(Registers.CONFIG_WLONG) : (byte)0 });
+        }
+
+        private void PowerOn()
+        {
+
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), (byte)Registers.ENABLE_PON });
+            Thread.Sleep(10);
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), (byte)(Registers.ENABLE_PON | Registers.ENABLE_AEN) });
+        }
+
+        private void PowerOff()
+        {
+            var powerState = I2cRead8(Registers.ENABLE);
+            powerState = (byte)(powerState & ~(byte)(Registers.ENABLE_PON | Registers.ENABLE_AEN));
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), powerState });
+        }
+
+        /// <summary>
+        /// Set/Clear the colors and clear interrupts
+        /// </summary>
+        /// <param name="state">true to set all interrupts, false to clear</param>
+        public void SetInterrupt(bool state)
+        {
+            SetInterrupt(InterruptState.All, state);
+        }
+
+        /// <summary>
+        /// Set/clear a specific interrupt persistence
+        /// </summary>
+        /// <param name="interupt">The percistence cycles</param>
+        /// <param name="state">True to set the interrupt, false to clear</param>
+        public void SetInterrupt(InterruptState interupt, bool state)
+        {
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.PERS), (byte)interupt });
+            var enable = I2cRead8(Registers.ENABLE);
+            enable = state ? enable |= (byte)Registers.ENABLE_AIEN : enable = (byte)(enable & ~(byte)Registers.ENABLE_AIEN);
+            _i2cDevice.Write(new byte[] { (byte)(Registers.COMMAND_BIT | Registers.ENABLE), enable });
+        }
+
+        /// <summary>
+        /// Get the color
+        /// </summary>
+        /// <param name="delay">Wait to read the data that the integration time is passed</param>
+        /// <returns></returns>
+        public Color GetColor(bool delay = true)
```

Comment:
I can add a property. The Reason it's a function is because you can force the reading withtout waiting (if you waited before for example for sopme reasons) or you just measure the data time to time.